sdks/python/apache_beam/internal/cloudpickle_pickler.py

```diff
@@ -29,16 +29,20 @@
 
 import base64
 import bz2
+import contextlib
 import io
 import logging
 import sys
 import threading
+from unittest import mock
```

Comment:
would it suffice to:

```
def _custom_dynamic_function_reduce():
   ...

if (<setting is enabled>):
   cloudpickle._dynamic_function_reduce = _custom_dynamic_function_reduce
```

so that we:
1) patch a single function
2) don't pull a test dependency (unittest.mock)

Reply:
Yes, I think we could forego the context manager and do the unconditional patching. I think that's sufficient for your usecase since for the lifetime of the process the setting is either enabled or disabled.

Reply:
import is no longer needed